integration/insert_compat_test.go

```diff
@@ -68,30 +74,43 @@ func testInsertCompat(t *testing.T, testCases map[string]insertCompatTestCase) {
 						compatErr = UnsetRaw(t, compatErr)
 
 						// Skip inserts that could not be performed due to Tigris schema validation.
-						var e mongo.CommandError
-						if errors.As(targetErr, &e) && e.Name == "DocumentValidationFailure" {
-							if e.HasErrorCodeWithMessage(121, "json schema validation failed for field") {
-								setup.SkipForTigrisWithReason(t, targetErr.Error())
-							}
+						var e mongo.BulkWriteException
+						if errors.As(targetErr, &e) &&
+							e.HasErrorCodeWithMessage(121, "json schema validation failed for field") {
+							setup.SkipForTigrisWithReason(t, targetErr.Error())
 						}
 
-						assert.Equal(t, compatErr, targetErr)
+						assert.Equal(t, compatErr.Error(), targetErr.Error())
 					} else {
 						require.NoError(t, compatErr, "compat error; target returned no error")
 					}
 
-					var targetFindRes, compatFindRes bson.D
-					targetCursor, err := targetCollection.Find(ctx, bson.D{{}})
+					if !tc.skipIDcheck {
+						require.Equal(t, compatInsertRes, targetInsertRes)
+					}
+
+					findOpts := options.Find().SetSort(bson.D{{"_id", 1}})
+
+					var targetFindRes, compatFindRes []bson.D
+					targetCursor, err := targetCollection.Find(ctx, bson.D{{}}, findOpts)
 					require.NoError(t, err)
 					defer targetCursor.Close(ctx)
-					targetCursor.Decode(&targetFindRes)
+					err = targetCursor.All(ctx, &targetFindRes)
+					require.NoError(t, err)
 
-					compatCursor, err := compatCollection.Find(ctx, bson.D{{}})
+					compatCursor, err := compatCollection.Find(ctx, bson.D{{}}, findOpts)
 					require.NoError(t, err)
 					defer compatCursor.Close(ctx)
-					compatCursor.Decode(&targetFindRes)
+					err = compatCursor.All(ctx, &compatFindRes)
```

Comment:
I added some more logic to find and compare data as there were not many actual checks in this test before.

Reply:
We could simplify it with `FindAll` helper

Reply:
(fixed)